src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs

```diff
@@ -32,27 +33,24 @@ public static IApplicationBuilder UseHttpsRedirection(this IApplicationBuilder a
 
             var options = app.ApplicationServices.GetRequiredService<IOptions<HttpsRedirectionOptions>>().Value;
 
-            // The tls port set in options will have priority over the one in configuration.
-            var httpsPort = options.HttpsPort;
-            if (httpsPort == null)
+            // Order for finding the HTTPS port:
+            // 1. Set in the HttpsRedirectionOptions
+            // 2. HTTPS_PORT environment variable
```

Comment:
This will mean that any reverse proxy will have to setup this environment variable, even when listening to 443 on the external address

Reply:
Actually, I had an idea for this. Can we make the forwarding headers middleware play along with this one (by putting a feature on the request) and have this middleware not use the server addresses feature if the request was forwarded? That way the logic will be:

// Order for finding the HTTPS port:
// 1. Set in the HttpsRedirectionOptions
// 2. HTTPS_PORT environment variable
// 3. if the request was not forwarded, IServerAddressesFeature
// 4. 443 (or not set)

Reply:
That's way too many dominoes. Let's not make this any more complicated than it needs to be.

Reading IServerAddressesFeature is only a problem if the local server has HTTPS it's not on the same port as the proxy. The average reverse proxy uses HTTP to communicate with the back-end.